common/djangoapps/student/models.py

```diff
@@ -1675,6 +1684,8 @@ def emit_event(self, event_name, enterprise_uuid=None, request=None):  # pylint:
                         }
                     )
                     optimizely_client.track('welcome_email_sent', str(self.user.id))
+                    if exception_raised:
```

Comment:
Can we add a check here for control so that we don't fire this event for control group.
```suggestion
                    if exception_raised and not optimizely_experiment_variation == 'redesign_email_enabled':
```

Reply:
Yes. We should add this condition because we are sending legacy email to control user group.

Reply:
But we can also check both counts(control and variation) for `welcome_email_not_sent` event in optimizely without adding this condition.

Reply:
yes but the event would make sense for control group because they actually do get the email even if there was an exception raised.